src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/PackageSpecFactory.cs

```diff
@@ -0,0 +1,947 @@
+                    TargetAlias = targetAlias,
+                    Warn = warn
+                };
+
+                targetFrameworkInfos.Add(targetFrameworkInformation);
+            }
+
+            return targetFrameworkInfos;
+        }
+
+        private static RestoreAuditProperties? GetRestoreAuditProperties(IProject project)
+        {
+            string enableAudit = project.OuterBuild.GetProperty("NuGetAudit");
+            string auditLevel = project.OuterBuild.GetProperty("NuGetAuditLevel");
+            string auditMode = GetAuditMode(project);
+            HashSet<string>? suppressionItems = GetAuditSuppressions(project.OuterBuild);
+
+            if (enableAudit != null || auditLevel != null || auditMode != null
+                || (suppressionItems != null && suppressionItems.Count > 0))
+            {
+                return new RestoreAuditProperties()
+                {
+                    EnableAudit = enableAudit,
+                    AuditLevel = auditLevel,
+                    AuditMode = auditMode,
+                    SuppressedAdvisories = suppressionItems?.Count > 0 ? suppressionItems : null
+                };
+            }
+
+            return null;
+
+            // We want to set NuGetAuditMode to "all" if a multi-targeting project targets .NET 10 or higher.
+            // However, that can only be done by an "inner build" evaulation, but we read other audit settings
+            // from the project evaluation, not inner-builds. So, check the inner builds if any TFM sets mode
+            // to "all", otherwise use the project's "outer build" mode.
+            string GetAuditMode(IProject project)
+            {
+                foreach (var item in project.TargetFrameworks.NoAllocEnumerate())
+                {
+                    string auditMode = item.Value.GetProperty("NuGetAuditMode");
+                    if (string.Equals(auditMode, "all", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return auditMode;
+                    }
+                }
+
+                string projectAuditMode = project.OuterBuild.GetProperty("NuGetAuditMode");
+                return projectAuditMode;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating if the specified project is a legacy project.
+        /// </summary>
+        /// <param name="project">The <see cref="ITargetFramework" /> representing the project.</param>
+        /// <returns><code>true</code> if the specified project is considered legacy, otherwise <code>false</code>.</returns>
+        internal static bool IsLegacyProject(ITargetFramework project)
+        {
+            // We consider the project to be legacy if it does not specify TargetFramework or TargetFrameworks
+            return project.GetProperty("TargetFramework") == null && project.GetProperty("TargetFrameworks") == null;
```

Comment:
This can be a future update, consider defining constants for all of these values in this class. Or use predefined ones if they exist.

Reply:
I intended to do that!  Oops, I forgot.

Reply:
The commit making the change is +389 -211 lines changed, so I'm not sure if the team would prefer it in this PR or a new PR:  https://github.com/NuGet/NuGet.Client/commit/e04a15f86d6d38dc166f6c7f6905d9e8a862641c

Reply:
Your call. Whatever makes it easier for you. 

This PR is already green, and I'm about to approve, and sounds you do have a bit of a preference :D